crates/legacy/src/service.rs

```diff
@@ -1235,15 +1778,3188 @@ pub(crate) async fn handle_received_txns<Types: NodeType>(
+                    panic!("Expected an error, but got a result");
+                }
+                None => {
+                    panic!("Expected an error, but got a result");
+                }
+            }
+        }
+
+        let mut tx_receiver = tx_receiver;
+        assert!(tx_receiver.next().await.is_some());
+        assert!(tx_receiver.next().await.is_some());
+        assert!(tx_receiver.next().await.is_none());
+    }
+
+    /// This test checks that the error [HandleReceivedTxnsError::TransactionTooBig]
+    /// when the conditions are met.
+    ///
+    /// To trigger this error we simply provide a [TestTransaction] whose size
+    /// exceeds the maximum transaction length. we pass to [HandleReceivedTxns].
+    #[async_std::test]
+    async fn test_handle_received_txns_error_transaction_too_big() {
+        let (tx_sender, tx_receiver) = async_broadcast::broadcast(10);
+        let num_transactions = 2;
+        let mut txns = Vec::with_capacity(num_transactions + 1);
+        for index in 0..num_transactions {
+            txns.push(TestTransaction::new(vec![index as u8]));
+        }
+        txns.push(TestTransaction::new(vec![0; 256]));
+        let txns = txns;
+
+        {
+            let mut handle_received_txns_iter = HandleReceivedTxns::<TestTypes>::new(
+                tx_sender,
+                txns.clone(),
+                TransactionSource::HotShot,
+                10,
+            );
+
+            assert!(handle_received_txns_iter.next().is_some());
+            assert!(handle_received_txns_iter.next().is_some());
+            match handle_received_txns_iter.next() {
+                Some(Err(HandleReceivedTxnsError::TransactionTooBig {
+                    estimated_length,
+                    max_txn_len,
+                })) => {
+                    // This is expected,
+                    assert!(estimated_length >= 256);
+                    assert_eq!(max_txn_len, 10);
+                }
+                Some(Err(err)) => {
+                    panic!("Unexpected error: {:?}", err);
+                }
+                Some(Ok(_)) => {
+                    panic!("Expected an error, but got a result");
+                }
+                None => {
+                    panic!("Expected an error, but got a result");
+                }
+            }
+        }
+
+        let mut tx_receiver = tx_receiver;
+        assert!(tx_receiver.next().await.is_some());
+        assert!(tx_receiver.next().await.is_some());
+        assert!(tx_receiver.next().await.is_none());
+    }
+
+    /// This test checks that the error [HandleReceivedTxnsError::Internal]
+    /// is returned when the broadcast channel is closed.
+    ///
+    /// To trigger this error we simply close the broadcast channel receiver
+    /// before attempting to send any transactions through the broadcast channel
+    /// sender.
+    #[async_std::test]
+    async fn test_handle_received_txns_error_something() {
```

Comment:
Nit: Why "something" here?

Reply:
Ah, this is a correction I missed.  Essentially an off-by-1 error.  When I start making the specific test to test for an error case, I'm not necessarily certain which error I'm testing for.  As a result, I use a temporary placeholder name.  In this case I failed to change the name so the temporary name became a permanent name.

Addressed in f15ede4c5e99fc012271066da8c4e6fba4ce3e13.